Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -13,10 +13,15 @@ private enum SearchListSection: Int {
     case Search
 }
 
+protocol SearchViewControllerDelegate: class {
```

Comment:
I didn't see your reply go by. Github's awful for this stuff.

Yeah, I just wanted a SiteView and SiteViewDelegate here to save us from having an entire delegate interface for nothing, and save all the HomePanels from caring that they're "HomePanels". i.e. they just become reusable ViewControllers that we can place where we want, when we want. It also kills off 4 interfaces and replaces them with two. It saves us from writing a HomePanelDelegate implementation and a SearchViewDelegate implementation that we have to keep in sync all the time because they do the exact same thing. It also let us reuse more code between the SiteTableViewControllers, saving us from differences popping up between them, but still letting new implementations have the freedom to do whatever they want.

I don't think I understand what problem you're trying to solve that you pointed to? What do we gain from keeping the SearchViewDelegate and HomePanelDelegate separate?